File: 2089-maximum-matrix-sum/maximum-matrix-sum.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& mat) {
       long long sum=0;
       int count=0;
       int mini=INT_MAX;
       for(int i=0;i<mat.size();i++)
       {
        for(int j=0;j<mat[0].size();j++)
        {
            sum+=abs(mat[i][j]);
            if(abs(mat[i][j])<mini)
            {
                mini=abs(mat[i][j]);
            }
            if(mat[i][j]<0)
            {
                ++count;
            }
        }
       }
       if(count%2==0)
       {
        return sum;
       }
       return sum-2*mini; 
    }
};

File: 2089-maximum-matrix-sum/maximum-matrix-sum.java
class Solution {
    public long maxMatrixSum(int[][] mat) {
     int count=0;
     long sum=0;
     long mini=1000000000;
     int n=mat.length;
     int m=mat[0].length;
     for(int i=0;i<n;i++)
     {
        for(int j=0;j<m;j++)
        {
            sum+=Math.abs(mat[i][j]);
            if(mat[i][j]<0)
            {
                ++count;
            }
            mini=Math.min(mini,Math.abs(mat[i][j]));
        }
     }
     if(count%2==0)
     {
        return sum;
     }
     if(count>0)
     {
        return sum-2*mini;
     }
     return sum;

    }
}